src/Symfony/Component/OptionsResolver/OptionsResolver.php

```diff
@@ -784,14 +790,20 @@ public function resolve(array $options = array())
      * @throws OptionDefinitionException If there is a cyclic dependency between
      *                                   lazy options and/or normalizers
      */
-    public function offsetGet($option)
+    public function offsetGet($option/*, bool $triggerDeprecation = true*/)
     {
         if (!$this->locked) {
             throw new AccessException('Array access is only supported within closures of lazy options and normalizers.');
         }
 
+        $triggerDeprecation = 1 === \func_num_args() || \func_get_arg(1);
+
         // Shortcut for resolved options
         if (array_key_exists($option, $this->resolved)) {
+            if ($triggerDeprecation && isset($this->deprecated[$option]) && \is_string($this->deprecated[$option])) {
```

Comment:
What happens when the deprecation is configured as a callback?

Reply:
Exactly, deprecation based on the value should be triggers only if the option is provided by the user. See test cases.